packages/user-interface/src/components/MenuItem.tsx

```diff
@@ -15,7 +16,15 @@ interface MenuItemProps {
 
 const MenuItem = ({ item, current = false }: MenuItemProps) => {
   const { hrefLang } = useContext(LocaleContext);
-  const { hideMenu, messagesCount } = useContext(LayoutContext);
+  const { hideMenu, messagesCount, setMessagesCount } =
+    useContext(LayoutContext);
+
+  useGetBerichtenQuery({
+    skip: !item.hasMessagesCount,
+    onCompleted: (data: { getBerichten: { totalElements: number } }) => {
+      setMessagesCount(data?.getBerichten?.totalElements || 0);
+    },
```

Comment:
Should we add some polling here to update once in a while? https://www.apollographql.com/docs/react/data/queries/#polling

Reply:
see comment above. 

but yes, good idea.